src/sql/workbench/contrib/searchViewPane/browser/searchWidget/searchWidgetService.ts

```diff
@@ -0,0 +1,114 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import { ViewPaneContainer } from 'vs/workbench/browser/parts/views/viewPaneContainer';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IInstantiationService, createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { Extensions as ViewContainerExtensions, IViewsRegistry, IViewDescriptorService } from 'vs/workbench/common/views';
+import { NotebookSearchResultsView } from 'sql/workbench/contrib/notebook/browser/notebookExplorer/searchResultsViewPane';
+import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors';
+import { localize } from 'vs/nls';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { MenuId } from 'vs/platform/actions/common/actions';
+import { SimpleSearchWidget, IViewExplorerSearchOptions } from 'sql/workbench/contrib/searchViewPane/browser/searchWidget/simpleSearchWidget';
+
+export const ISearchWidgetService = createDecorator<ISearchWidgetService>('searchWidgetService');
+
+export interface ISearchViewPaneOptions {
+	VIEWLET_ID: string;
+	actionsMenuId: MenuId;
+	showSearchResultsPane: boolean;
+	onSearchSubmit(options: any): void;
+	onSearchCancel({ focus: boolean }): void;
+}
+export interface ISearchWidgetService {
+	searchWidgets: Map<string, any>;
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void;
+	getSearchWidget(VIEWLET_ID: string): any | undefined;
+	getSearchResultsView(parentConatiner: ViewPaneContainer): any | undefined;
+}
+export class SearchWidgetService extends Disposable implements ISearchWidgetService {
+
+	declare readonly _serviceBrand: undefined;
+	searchWidgets: Map<string, any>;
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IViewDescriptorService protected viewDescriptorService: IViewDescriptorService,
+	) {
+		super();
+		this.searchWidgets = new Map<string, any>();
+	}
+
+	getSearchWidget(VIEWLET_ID: string): any | undefined {
+		return this.searchWidgets.get(VIEWLET_ID);
+	}
+
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void {
+		if (!this.searchWidgets.get(params.VIEWLET_ID)) {
+			let searchWidget = this._register(this._instantiationService.createInstance(SimpleSearchWidget, container, <IViewExplorerSearchOptions>{
+				value: '',
+				replaceValue: undefined,
+				isRegex: false,
+				isCaseSensitive: false,
+				isWholeWords: false,
+				searchHistory: [],
+				replaceHistory: [],
+				preserveCase: false,
+				showSearchResultsPane: params.showSearchResultsPane,
+			}, pane, params.VIEWLET_ID));
+
+			this.searchWidgets.set(params.VIEWLET_ID, searchWidget);
+			if (params.showSearchResultsPane) {
+				this.registerNotebookSerachResultsView(params);
```

Comment:
Spelling mistake here on registerNotebookSearchResultsView

But also why are we referencing notebook stuff here?